lib/index.js

```diff
@@ -10,23 +10,28 @@ const getGUID = () => {
   })
 }
 
+const must = required => {
+  Object.keys(required).forEach(key => {
+    if (required[key] === undefined) {
+      throw Error(`'${key} option is required`)
+    }
+  })
+}
+
 class Sharepoint {
   constructor (url) {
-    if (!url) {
-      throw new Error('You must provide a url.')
-    }
+    must({ url })
 
     this.url = url
     this.headers = null
     this.site = null
   }
 
-  async authenticate (username, password) {
-    if (!username && !password) {
-      throw new Error('You must provide a username and password.')
+  async authenticate (credentials) {
+    if (!credentials) {
+      throw Error('Credentials are required')

```

Comment:
I removed all `new` before Error because it doesn't do anything in Javascript really other than creating an unnecessary object in memory maybe.